pythclient/hermes.py

```diff
@@ -0,0 +1,173 @@
+import asyncio
+from typing import TypedDict
+
+import httpx
+import os
+
+from .price_feeds import Price
+
+HERMES_ENDPOINT_HTTPS = "https://hermes.pyth.network/api/"
+HERMES_ENDPOINT_WSS = "wss://hermes.pyth.network/ws"
+
+
+class PriceFeed(TypedDict):
+    feed_id: str
+    price: Price
+    ema_price: Price
+    vaa: str
+
+
+
+class HermesClient:
+    def __init__(self, feed_ids: list[str], endpoint=HERMES_ENDPOINT_HTTPS, ws_endpoint=HERMES_ENDPOINT_WSS):
+        self.feed_ids = feed_ids
+        self.pending_feed_ids = feed_ids
+        self.prices_dict: dict[str, PriceFeed] = {}
+        self.client = httpx.AsyncClient()
+        self.endpoint = endpoint
+        self.ws_endpoint = ws_endpoint
+
+    async def get_price_feed_ids(self) -> list[str]:
+        """
+        Queries the Hermes https endpoint for a list of the IDs of all Pyth price feeds.
+        """
+
+        url = os.path.join(self.endpoint, "price_feed_ids")
+        
+        client = httpx.AsyncClient()
+
+        data = (await client.get(url)).json()
+
+        return data
+
+    def add_feed_ids(self, feed_ids: list[str]):
+        self.feed_ids += feed_ids
+        self.feed_ids = list(set(self.feed_ids))
+        self.pending_feed_ids += feed_ids
```

Comment:
why does `self.pending_feed_ids` not require uniqueness?

Reply:
we always call `list(set(self.feed_ids))`, so the feed_ids will always be unique

Reply:
it's possible for `self.pending_feed_ids` to contain duplicates if the same feed IDs are added more than once before they are cleared, for e.g. 
- `add_feed_ids` method is called with `['feed1', 'feed2']`
- before any operation that clears `self.pending_feed_ids` is performed, `add_feed_ids` is called again with ['feed2', 'feed3']
- `self.feed_ids` will now contain ['feed1', 'feed2', 'feed3'] without duplicates because it's converted to a set and then back to a list
- however, `self.pending_feed_ids` will contain ['feed1', 'feed2', 'feed2', 'feed3'], where 'feed2' is duplicated.

Reply:
imo you can do something like this to make it cleaner
```
    # Convert feed_ids to a set to remove any duplicates from the input
    new_feed_ids_set = set(feed_ids)
    
    # Update self.feed_ids; convert to set for union operation, then back to list
    self.feed_ids = list(set(self.feed_ids).union(new_feed_ids_set))
    
    # Update self.pending_feed_ids with only those IDs that are truly new
    self.pending_feed_ids = list(set(self.pending_feed_ids).union(new_feed_ids_set))
```

Reply:
another thing that we can possibly do here is to validate that the `feed_ids` passed in is in the format we expect
e.g.
```
def validate_feed_ids(self, feed_ids: list[str]):
    """
    Validates the format of feed IDs. Each ID should be a 64-character hexadecimal string,
    optionally prefixed with '0x'.
    """
    hex_pattern = re.compile(r'^0x[a-fA-F0-9]{64}$|^[a-fA-F0-9]{64}$')
    for feed_id in feed_ids:
        if not hex_pattern.match(feed_id):
            raise ValueError(f"Invalid feed ID format: {feed_id}")
```
and then call `validate_feed_ids` in the first line of `add_feed_ids` and write tests to ensure that invalid ids (e.g. 30 chars, invalid chars, etc) throw an error